server/service/scenarios/middleware.js

```diff
@@ -0,0 +1,37 @@
+const { asyncMiddleware } = require('../../util/api');
+const db = require('./db');
+
+const scenarioMap = new WeakMap();
+
+const defaultIdParam = req => Number(req.params.scenario_id);
+
+exports.reqScenario = req => {
+    if (!scenarioMap.has(req)) {
+        throw new Error(
+            'Request has not passed through lookupScenario middleware'
+        );
+    }
+    return scenarioMap.get(req);
+};
+
+const scenarioForRequest = async (req, getId = defaultIdParam) => {
```

Comment:
Why rename `defaultIdParam`?  Couldn't we just name it `getId` initially and avoid this step?

Reply:
Oh actually never mind, I get it now. We could pass in another callback at another point if we need to.

Reply:
yeah the default grabs it from the route params - but it could be as deep as something like `req => req.body.scenario_id` or something for some secondary endpoint or especially for an endpoint in another router

Reply:
Cool! I like that idea